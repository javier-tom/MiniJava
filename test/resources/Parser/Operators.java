class Main {
    public static void main(String[] args) {
        System.out.println(new Operators().Ops());
    }
}

class Operators {
    public int Ops() {
        int i;
        i = 0;
        i = i + 5;
        i = i * 3;
        i = i - 10;
        i = true && false;
        i = !true;
        i = true < false;
        return 0;
    }
}