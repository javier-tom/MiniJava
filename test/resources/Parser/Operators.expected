Program
  MainClass Main (line 1)
    Print (line 3)
      Call Ops (line 3)
        on class:
          NewObject Operators (line 3)
        parameters:

  Class Operators (line 7)
    fields:
    MethodDecl Ops (line 8)
      returns int
      params:
      variables:
        VarDecl int i (line 9)
      Assign i (line 10)
        new value:
          0 (line 10)
      Assign i (line 11)
        new value:
          Plus (line 11)
            i (line 11)
            5 (line 11)
      Assign i (line 12)
        new value:
          Times (line 12)
            i (line 12)
            3 (line 12)
      Assign i (line 13)
        new value:
          Minus (line 13)
            i (line 13)
            10 (line 13)
      Assign i (line 14)
        new value:
          And (line 14)
            True (line 14)
            False (line 14)
      Assign i (line 15)
        new value:
          Not (line 15)
            True (line 15)
      Assign i (line 16)
        new value:
          LessThan (line 16)
            True (line 16)
            False (line 16)
      Return
        0 (line 17)